### PR TITLE
feat: asset storage

### PR DESCRIPTION
**Pull request by fgerschwiler**

### CODE REVIEW

File: cmd/zitadel/startup.yaml [hunk 1]
```diff
@@ -11,6 +11,15 @@ Tracing:
...
 
+AssetStorage:
+  Type: $ZITADEL_ASSET_STORAGE_TYPE
+  Config:
+    Endpoint: 'play.min.io'
+    AccessKeyID: 'Q3AM3UQ867SPQQA43P2F'
+    SecretAccessKey: 'zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG'
+    SSL: TRUE
+    Location: 'us-east-1'
+
```
Comment (fforootd) [inline cmd/zitadel/startup.yaml:22]: we should remove this values 😁

Comment (fgerschwiler) [reply cmd/zitadel/startup.yaml:22]: these are not valid anymore, but I will remove it. I already have some ENV_VARs for this

Comment (fforootd) [reply cmd/zitadel/startup.yaml]: 👍

**Comment by fforootd** [comment]
i would like to rename this PR to `feat: asset storage` instead of s3.

Any objections?

File: internal/eventstore/repository/repository.go [hunk 2]
```diff
@@ -12,7 +12,7 @@ type Repository interface {
 	// if assets are pushed, they will be added to the assets table only referenced by id in eventstore (history)
```
Comment (livio-a) [inline internal/eventstore/repository/repository.go:12]:
```suggestion
```

File: internal/static/s3/minio.go [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+	Location string
+}
+
+func NewMinio(config S3Config) (*Minio, error) {
+	minioClient, err := minio.New(config.Endpoint, &minio.Options{
+		Creds:  credentials.NewStaticV4(config.AccessKeyID, config.SecretAccessKey, ""),
+		Secure: config.SSL,
+		Region: config.Location,
+	})
+	if err != nil {
```
Comment (livio-a) [inline internal/static/s3/minio.go:29]: IMHO we should wrap the errors with the caos error pkg so we don't have to handle specific errors in packages using the static package

Comment (livio-a) [reply internal/static/s3/minio.go:29]: like
```
return caos_errs.ThrowAlreadyExists(nil, "MINIO-9n3MK", "Errors.Assets.Bucket.AlreadyExists")
```

File: internal/static/s3/minio.go [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+	return presignedURL, err
+}
+
+func (m *Minio) ListObjectInfos(ctx context.Context, bucketName, prefix string) ([]*domain.AssetInfo, error) {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	objectCh := m.Client.ListObjects(ctx, bucketName, minio.ListObjectsOptions{
+		Prefix:    prefix,
+		Recursive: true,
```
Comment (livio-a) [inline internal/static/s3/minio.go:115]: do we always need `recursive`?

File: internal/static/s3/minio.go [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+	})
+	assetInfos := make([]*domain.AssetInfo, 0)
+	for object := range objectCh {
+		if object.Err != nil {
+			logging.LogWithFields("MINIO-wC8sd", "bucket-name", bucketName, "prefix", prefix).WithError(object.Err).Debug("unable to ger object")
```
Comment (livio-a) [inline internal/static/s3/minio.go:120]:
```suggestion
			logging.LogWithFields("MINIO-wC8sd", "bucket-name", bucketName, "prefix", prefix).WithError(object.Err).Debug("unable to get object")
```

**Review by livio-a — APPROVED** [review]

**Pull request merged**